hls4ml/writer/vivado_writer.py

```diff
@@ -305,6 +305,8 @@ def write_defines(self, model):
                 newline = line
                 numbers = OrderedDict.fromkeys([layer.get_numbers_cpp() for layer in model.get_layers()])
                 newline += ''.join(numbers)
+                newline = '\n'.join(set(newline.split('\n')))
```

Comment:
try `newline += ''.join(set(numbers))` on the previous line instead (or something like that)